packages/mapper/src/mapper.ts

```diff
@@ -495,18 +609,31 @@ export default class Mapper {
    */
   async findOneAndUpdate(collection, filter, update, options = {}): Promise<any> {
     const findOneAndUpdateOptions: any = { ...options };
-    this.messageBus.emit('method:findOneAndUpdate', collection._name, filter);
+    const db = collection._database._name;
+    const coll = collection._name;
 
     if ('returnNewDocument' in findOneAndUpdateOptions) {
       findOneAndUpdateOptions.returnDocument = findOneAndUpdateOptions.returnNewDocument;
       delete findOneAndUpdateOptions.returnNewDocument;
     }
+
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'findOneAndUpdate',
+        class: 'Collection',
+        db,
+        coll,
+        arguments: { filter, update, findOneAndUpdateOptions }
```

Comment:
same as with `replacement` above: do we want to log the `update` doc?

Reply:
i start to see that could happen everywhere :/, In case of some import script we would have log files bigger than the database

Reply:
yes, exactly! that's why I want to be careful o.O